Chaincase.Common/Global.cs

```diff
@@ -260,6 +246,8 @@ public async Task InitializeNoWalletAsync(CancellationToken cancellationToken =
                 #endregion SynchronizerInitialization
 
                 TransactionBroadcaster = new TransactionBroadcaster(Network, BitcoinStore, Synchronizer, Nodes, WalletManager, null);
+                // CoinJoinProcessor maintains an event handler to process joins.
+                // We need this reference.
                 CoinJoinProcessor = new CoinJoinProcessor(Synchronizer, WalletManager, null);
```

Comment:
this can also be a separate service?  It doesn't seem to be directly used in here. You can make it require Global in the DI to resolve its current dependencies

Reply:
@armins88 and I were looking at this the other day. It's a pain cause it needs to be constructed and just not disposed. If we make a singleton in DI I don't know if it's constructed or needs to be called and then it's lazy loaded. Do you know?